dm-manager/pkg/dm/amt.go

```diff
@@ -5,65 +5,333 @@ package dm
 
 import (
 	"context"
+	"fmt"
+	"strings"
 	"sync"
 	"time"
 
+	inventoryv1 "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/inventory/v1"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/client"
 	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/logging"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/secretprovider"
 	"github.com/open-edge-platform/infra-external/dm-manager/pkg/api/mps"
+	"github.com/open-edge-platform/infra-external/dm-manager/pkg/api/rps"
+)
+
+const (
+	mpsPort          = 4433
+	fqdnServerFormat = 201
+	passwordAuth     = 2
+
+	AmtPasswordSecretName = "amt-password"
+	passwordKey           = "password"
+
+	StaticPasswordPolicy  = "static"
+	DynamicPasswordPolicy = "dynamic"
 )
 
 var log = logging.GetLogger("DmReconciler")
 
+type ReconcilerConfig struct {
+	ClusterDomain  string
+	PasswordPolicy string
+
+	RequestTimeout  time.Duration
+	ReconcilePeriod time.Duration
+}
 type Reconciler struct {
-	APIClient mps.ClientWithResponsesInterface
-	TermChan  chan bool
-	ReadyChan chan bool
-	WaitGroup *sync.WaitGroup
+	MpsClient       mps.ClientWithResponsesInterface
+	RpsClient       rps.ClientWithResponsesInterface
+	InventoryClient client.TenantAwareInventoryClient
+	EventsWatcher   chan *client.WatchEvents
+	SecretProvider  secretprovider.SecretProvider
+	TermChan        chan bool
+	ReadyChan       chan bool
+	WaitGroup       *sync.WaitGroup
+	Config          *ReconcilerConfig
 }
 
 func (dmr *Reconciler) Start() {
-	ticker := time.NewTicker(time.Minute)
+	ticker := time.NewTicker(dmr.Config.ReconcilePeriod)
 	if dmr.ReadyChan != nil {
 		dmr.ReadyChan <- true
 	}
 	log.Info().Msgf("Starting periodic reconciliation")
-	dmr.Reconcile(context.Background())
+	dmr.ReconcileAdd()
+	dmr.ReconcileRemove()
```

Comment:
can we have a single function? And we should handle the errors possibly. I dont see why we need this logical split in 2 functions